pyanaconda/geoloc.py

```diff
@@ -117,205 +124,194 @@
-    # use the Fedora GeoIP provider
-    default_provider = FedoraGeoIPProvider
-    provider = providers.get(provider_id, default_provider)
-    return provider()
+        None might be returned if:
 
+        - no results were found
+        - the refresh is still in progress
 
-def _get_location_info_instance(wait=False):
-    """
-    Return instance of the location info object
-    and optionally wait for the Geolocation thread to finish
+        :return: :class:LocationResult instance or None if location is unknown
+        :rtype: :class:LocationResult instance or None
+        """
+        return self._location_info.result
 
-    If there is no lookup in progress (no Geolocation refresh thread
-    is running), this function returns at once).
+    def _get_provider_id_from_option(self, option_string):
+        """Get a valid provider id from a string.
 
-    Meaning of the wait parameter:
-    - False or <=0: don't wait
-    - True : wait for default number of seconds specified by
-    the GEOLOC_TIMEOUT constant
-    - >0 : wait for a given number of seconds
+        This function is used to parse command line
+        arguments/boot options for the geolocation module.
 
-    :param wait: specifies if this function should wait
-    for the lookup to finish before returning the instance
-    :type wait: bool or integer or float
-    """
-    if not wait:
-        # just returns the instance
-        return location_info_instance
-
-    # check if wait is a boolean or a number
-    if wait is True:
-        # use the default waiting period
-        wait = constants.GEOLOC_TIMEOUT
-    # check if there is a refresh in progress
-    start_time = time.time()
-    refresh_condition.acquire()
-    if refresh_in_progress:
-        # calling wait releases the lock and blocks,
-        # after the thread is notified, it unblocks and
-        # reacquires the lock
-        refresh_condition.wait(timeout=wait)
-        if refresh_in_progress:
-            log.info("Waiting for Geolocation timed out after %d seconds.", wait)
-            # please note that this does not mean that the actual
-            # geolocation lookup was stopped in any way, it just
-            # means the caller was unblocked after the waiting period
-            # ended while the lookup thread is still running
+        :param str option_string: option specifying the provider
+        :return: provider id
+        """
+        # normalize the option string, just in case
+        option_string = option_string.lower()
+        if option_string.lower() in OFFICIALLY_SUPPORTED_GEOLOCATION_PROVIDER_IDS:
+            return option_string
         else:
-            elapsed_time = time.time() - start_time
-            log.info("Waited %1.2f seconds for Geolocation", elapsed_time)
-    refresh_condition.release()
-    return location_info_instance
+            # fall back to the default provider
+            return None
 
 
 class GeolocationError(Exception):
```

Comment:
Please move this exception class to the top. AFAICT it is convention around here.